warehouse/helpdesk/services.py

```diff
@@ -120,3 +124,32 @@ def create_conversation(self, *, request_json: dict) -> str:
         resp.raise_for_status()
         # return the API-friendly location of the conversation
         return resp.headers["Location"]
+
+    def add_tag(self, *, conversation_url: str, tag: str) -> None:
+        """
+        Add a tag to a conversation in HelpScout
+        https://developer.helpscout.com/mailbox-api/endpoints/conversations/tags/update/
+        """
+        # Get existing tags and append new one
+        resp = self.http.get(
+            conversation_url, headers={"Authorization": f"Bearer {self.bearer_token}"}
+        )
+        resp.raise_for_status()
+
+        # collect tag strings from response
+        tags = [tag["tag"] for tag in resp.json()["tags"]]
+
+        if tag in tags:
+            # tag already exists, no need to add it
+            return
+
+        tags.append(tag)
+
+        resp = self.http.put(
+            f"{conversation_url}/tags",
+            headers={"Authorization": f"Bearer {self.bearer_token}"},
+            json={"tags": tags},
+            timeout=10,
+        )
+        resp.raise_for_status()
+        return
```

Comment:
Driveby comment, but based on my read of https://api.helpdesk.com/docs#operation/ticketUpdateAddTag it looks like this could just be:

```suggestion
        resp = self.http.put(
            f"{conversation_url}/tags",
            headers={"Authorization": f"Bearer {self.bearer_token}"},
            json={"tags": [tag]},
            timeout=10,
        )
        resp.raise_for_status()
        return
````

I suspect the API doesn't drop tags that already exist but aren't included in the request.

Reply:
@di Thanks, but that's the API docs for Help Desk, not Help Scout.